enhancements/network/allow-from-router-networkpolicy.md

```diff
@@ -0,0 +1,236 @@
+     `openshift-host-network` the same way (see below for more
+     details).
+
+  3. CNO will watch the ingress configuration, and when the endpoint
+     publishing strategy is `HostNetwork`, it will add the
+     `network.openshift.io/policy-group: ingress` label to
+     `openshift-host-network`, and when it's not `HostNetwork` it will
+     remove that label.
+
+Then users will be able to match router traffic in either
+openshift-sdn or ovn-kubernetes with any router endpoint publishing
+strategy via:
+
+    ingress:
+      - from:
+          - namespaceSelector:
+              matchLabels:
+                network.openshift.io/policy-group: ingress
+
+We can also formally specify "allow-from-host-network". (This is
+basically free at this point, and also useful to users, so...) We just
+have CNO also add the label `network.openshift.io/policy-group:
+host-network` to `openshift-host-network`, and then we document that
+(at least when using openshift-sdn or ovn-kubernetes), a NetworkPolicy
+selecting namespaces with that label will match host-network traffic.
+
+#### OpenShift SDN Modifications
+
+We _could_ avoid modifying openshift-sdn, and just have CNO know that
+the "host network" namespace is `openshift-host-network` with
+ovn-kubernetes, but `default` with openshift-sdn. However, it would be
+simpler if we just always used the same namespace as the "host network
+namespace".
+
+In NetworkPolicy mode, openshift-sdn doesn't let you have two
+namespaces with the same VNID, but we could easily hack it so that
+`openshift-host-network` was treated as VNID 0 _for purposes of
+NetworkPolicy only_. This would cause it to behave strangely if there
+were any pods in it, but we don't really care, since there _won't_ be
+any pods in it.
+
+### Risks and Mitigations
+
+The biggest problem is that there is really no good way to distinguish
+"traffic that's actually from a host-network router" from "random
+other host-network traffic", so users might end up allowing more
+traffic than they want to. This is mostly a documentation problem,
+since there really isn't a workaround if they are using `HostNetwork`.
+
+### Test Plan
+
+We will add ovn-kubernetes-specific tests for the new ovn-kubernetes
+feature upstream.
+
+We will add tests to origin to ensure that the
+`network.openshift.io/policy-group: host-network` label works as
+described above, when using openshift-sdn or ovn-kubernetes.
+
+We should also test that the `policy-group: ingress` label is applied
+correctly in `HostNetwork` endpoint publishing mode, and that
+NetworkPolicies based on that label work correctly regardless of
+endpoint publishing mode. I'm not sure if we currently run the origin
+e2e suite with `HostNetwork` endpoint publishing anywhere though...
+
+### Graduation Criteria
+
+##### Tech Preview -> GA 
+
+- Just one release of it being vaguely useful seems like enough? It's
+  not that complicated a feature...
+
+### Upgrade / Downgrade Strategy
```

Comment:
In context of openshift-sdn if we decide to support this, the downgrade needs to go back to supporting this how we do it today i.e. using the `default` namespace.

Reply:
The plan is not to change the handling of `default`. (That would essentially be an API break.) The plan is that NetworkPolicies selecting `default` and NetworkPolicies selecting `openshift-host-network` would _both_ select host network traffic. So any pre-existing policies selecting `default` would continue to work across the upgrade, and would likewise continue to work across a downgrade.

If the user _changed _their existing NetworkPolicies when upgrading then they'd have to change them back when downgrading, but that would be something the user would have to do (since they're the one who changed them in the first place).